addons/mail/static/src/messaging/entity/entity_core.js

```diff
@@ -0,0 +1,246 @@
+                );
+            }
+            if (!relation.to) {
+                throw new Error(
+                    `Relation "${
+                        Entity.name
+                    }/${
+                        relationName
+                    }" must define an Entity class name in "to".`
+                );
+            }
+            const RelatedEntity = Entities[relation.to];
+            if (!RelatedEntity) {
+                throw new Error(
+                    `Entity class name of relation "${Entity.name}/${relationName}" does not exist.`
+                );
+            }
+            const inverseRelation = RelatedEntity.relations[relation.inverse];
+            if (!inverseRelation) {
+                throw new Error(
+                    `Relation entity class "${
+                        Entity.name
+                    }/${
+                        relationName
+                    }" has no inverse relation "${RelatedEntity.name}/${relation.inverse}".`
+                );
+            }
+            const allSelfAndParentNames = [];
+            let target = Entity;
+            while (target) {
+                allSelfAndParentNames.push(target.name);
+                target = target.__proto__;
+            }
+            if (!allSelfAndParentNames.includes(inverseRelation.to)) {
+                throw new Error(
+                    `Relation "${
+                        Entity.name
+                    }/${
+                        relationName
+                    }" has inverse relation "${
+                        RelatedEntity.name
+                    }/${
+                        relation.inverse
+                    }" misconfigured (currently "${
+                        inverseRelation.to
+                    }", should instead refer to this entity or parented entity: ${
+                        allSelfAndParentNames.map(name => `"${name}"`).join(', ')
+                    }?)`
+                );
+            }
+            if (
+                (relation.type === 'many2many' && inverseRelation.type !== 'many2many') ||
+                (relation.type === 'one2one' && inverseRelation.type !== 'one2one') ||
+                (relation.type === 'one2many' && inverseRelation.type !== 'many2one') ||
+                (relation.type === 'many2one' && inverseRelation.type !== 'one2many')
+            ) {
+                throw new Error(
+                    `Mismatch relations types "${
+                        Entity.name
+                    }/${
+                        relationName
+                    }" (${
+                        relation.type
+                    }) and "${
+                        RelatedEntity.name
+                    }/${
+                        relation.inverse
+                    }" (${
+                        inverseRelation.type
+                    }).`
+                );
+            }
+        }
+    }
+}
+
+/**
+ * @returns {Object}
+ */
```

Comment:
Should document in details the goal of the method.
And maybe also inline comments to explain each important step.